components/webgpu/lib.rs

```diff
@@ -286,6 +300,50 @@ impl WGPU {
                         )
                     }
                 },
+                WebGPURequest::CreateShaderModule(sender, device, id, program) => {
+                    let global = &self.global;
+                    let descriptor = wgpu_core::pipeline::ShaderModuleDescriptor {
+                        code: wgpu_core::U32Array {
+                            bytes: program.as_ptr(),
+                            length: program.len(),
+                        },
+                    };
+                    let sm_id = gfx_select!(id => global.device_create_shader_module(device.0, &descriptor, id));
+                    let shader_module = WebGPUShaderModule(sm_id);
+
+                    if let Err(e) = sender.send(shader_module) {
+                        warn!(
+                            "Failed to send response to WebGPURequest::CreateShaderModule ({})",
+                            e
+                        )
+                    }
+                },
+                WebGPURequest::CreateComputePipeline(
+                    sender,
+                    device,
+                    id,
+                    layout,
+                    program,
+                    entry,
+                ) => {
+                    let global = &self.global;
+                    let descriptor = wgpu_core::pipeline::ComputePipelineDescriptor {
+                        layout,
+                        compute_stage: wgpu_core::pipeline::ProgrammableStageDescriptor {
+                            module: program,
+                            entry_point: entry.as_ptr(),
```

Comment:
what makes it null-terminated?

Reply:
I've created a `CString` from the entry_point which will be null-terminated then passed it as pointer to the descriptor.